tests/common-tests.h

```diff
@@ -20,6 +20,9 @@
 #include "soci/boost-fusion.h"
 #endif // BOOST_VERSION
 #endif // SOCI_HAVE_BOOST
+#if defined(_WIN32) && _MSVC_LANG > 201402L || __cplusplus > 201402L
```

Comment:
I think we ought to define `SOCI_HAS_CXX11` once in some header instead of repeating this unwieldy expression multiple times.

Reply:
I just added this define. It is currently not used.
I am not sure how to create such a define in cmake depending on the other cmake variables.

Reply:
You don't need to do it in CMake, you can just use exactly the same condition as above in some SOCI header (`soci-platform.h` probably).